fraud/service.go

```diff
@@ -88,20 +116,19 @@ func (f *service) processIncoming(
 		}
 		return pubsub.ValidationReject
 	}
+	// we can skip incoming proof if it has been already stored locally
+	if f.verifyLocal(ctx, proofType, proof.HeaderHash(), msg.Data) {
+		return pubsub.ValidationIgnore
+	}
+
+	msg.ValidatorData = proof
 
-	newCtx, cancel := context.WithTimeout(ctx, fetchHeaderTimeout)
-	extHeader, err := f.getter(newCtx, proof.Height())
-	defer cancel()
+	extHeader, err := f.getter(ctx, proof.Height())
 	if err != nil {
-		// Timeout means there is a problem with the network.
-		// As we cannot prove or discard Fraud Proof, user must restart the node.
-		if errors.Is(err, context.DeadlineExceeded) {
-			log.Errorw("failed to fetch header. Timeout reached.")
-			// TODO(@vgonkivs): add handling for this case. As we are not able to verify fraud proof.
-		}
+		// TODO @vgonkivs: add retry mechanism to fetch header
```

Comment:
Is there an issue that has been created for this?

Reply:
Not yes

Reply:
I think we can remove this TODO. The retry mechanism shouldn't be responsibility of fraud service but of the actual HeaderGetter impl